core/src/main/java/org/apache/iceberg/rest/requests/PlanTableScanRequestParser.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.util.List;
+import org.apache.iceberg.expressions.Expression;
+import org.apache.iceberg.expressions.ExpressionParser;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.JsonUtil;
+
+public class PlanTableScanRequestParser {
+  private static final String SNAPSHOT_ID = "snapshot-id";
+  private static final String SELECT = "select";
+  private static final String FILTER = "filter";
+  private static final String CASE_SENSITIVE = "case-sensitive";
+  private static final String USE_SNAPSHOT_SCHEMA = "use-snapshot-schema";
+  private static final String START_SNAPSHOT_ID = "start-snapshot-id";
+  private static final String END_SNAPSHOT_ID = "end-snapshot-id";
+  private static final String STATS_FIELDS = "stats-fields";
+
+  private PlanTableScanRequestParser() {}
+
+  public static String toJson(PlanTableScanRequest request) {
+    return toJson(request, false);
+  }
+
+  public static String toJson(PlanTableScanRequest request, boolean pretty) {
+    return JsonUtil.generate(gen -> toJson(request, gen), pretty);
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public static void toJson(PlanTableScanRequest request, JsonGenerator gen) throws IOException {
+    Preconditions.checkArgument(null != request, "Invalid plan table scan request: null");
+
+    gen.writeStartObject();
+    if (request.snapshotId() != null) {
+      gen.writeNumberField(SNAPSHOT_ID, request.snapshotId());
+    }
+
+    if (request.startSnapshotId() != null) {
+      gen.writeNumberField(START_SNAPSHOT_ID, request.startSnapshotId());
+    }
+
+    if (request.endSnapshotId() != null) {
+      gen.writeNumberField(END_SNAPSHOT_ID, request.endSnapshotId());
+    }
```

Comment:
On second look, the way this is written is a bit confusing after reading again since on the surface it seems like it's possible serializing snapshotId/startingSnapshotId/endSnapshotID even though it's either a point in time scan or an incremental scan.  To be clear, I know that validate checks this, I'm more so talking about how a reader would interpret this code.

concretely I'd recommend replacing this with:

````
private void serializeSnapshotIdForScan(JsonGenerator gen, request) {
    if (request.snapshotId() != null) {
         gen.writeNumberField(SNAPSHOT_ID, request.snapshotId());
     } else {
        gen.writeNumberField(START_SNAPSHOT_ID, request.startSnapshotId());
        gen.writeNumberField(END_SNAPSHOT_ID, request.endSnapshotId());
     }
}
````

I think then you'd also be able to get rid of cyclomatic complexity override above.

Reply:
Will follow this suggestion thanks @amogh-jahagirdar

Reply:
Actually when looking at this again are we sure this is what we want? When reading the spec again https://github.com/apache/iceberg/blob/main/open-api/rest-catalog-open-api.yaml#L609 
```
  If the request does not include either incremental or
        point-in-time config properties, scan planning should produce a
        point-in-time scan of the latest snapshot in the table's main branch
```

It seems possible that the client does not need to pass either a snapshotId, or start and end snapshot ids, and that the server can just plan with the latest snapshot id if nothing was provided. I would assume then at serialize time this would be an issue with the following code, 

```
private void serializeSnapshotIdForScan(JsonGenerator gen, request) {
    if (request.snapshotId() != null) {
         gen.writeNumberField(SNAPSHOT_ID, request.snapshotId());
     } else {
        gen.writeNumberField(START_SNAPSHOT_ID, request.startSnapshotId());
        gen.writeNumberField(END_SNAPSHOT_ID, request.endSnapshotId());
     }
}

```
as it may hit an error if the start and end ids were also null. I think the way it currently works because we the null check around each of the ids. Let me know what you think though